src/st/nz0/30958.c

```diff
@@ -78,8 +78,73 @@ void EntityRedEyeBust(Entity* self) {
     }
 }
 
-// DECOMPME_WIP func_801B12E8 https://decomp.me/scratch/3Sf68
-INCLUDE_ASM("asm/us/st/nz0/nonmatchings/30958", func_801B12E8);
+// A purplish-red brick background that scrolls behind the foreground layer
+void EntityPurpleBrickScrollingBackground(Entity* self) {
+    Primitive* prim;
+    s16 firstPrimIndex;
+    s32 tempPosX;
+    s32 tempPosY;
+    s32 x, y;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(&D_80180BF8);
+        self->posX.i.hi = 0;
+        self->posY.i.hi = 0;
+        self->unk68 = 0x80;
+        // Composed of 15 primitives
+        firstPrimIndex = g_api.AllocPrimitives(4, 15);
+        if (firstPrimIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+        prim = &g_PrimBuf[firstPrimIndex];
+        self->firstPolygonIndex = (s32)firstPrimIndex;
+        *(s32*)&self->unk7C = prim;
+        self->flags |= 0x800000;
+        if (prim != NULL) {
+            do {
+                prim->tpage = 0xF;
+                prim->clut = 4;
+                prim->u2 = 0x80;
+                prim->u0 = 0x80;
+                prim->u3 = 0xFF;
+                prim->u1 = 0xFF;
+                prim->v1 = 0x80;
+                prim->v0 = 0x80;
+                prim->v3 = 0xBF;
+                prim->v2 = 0xBF;
+                prim->priority = 0x20;
+                prim->blendMode = 2;
+                prim = prim->next;
+            } while (prim != NULL);
+        }
```

Comment:
```suggestion
while (prim != NULL) {
                prim->tpage = 0xF;
                prim->clut = 4;
                prim->u2 = 0x80;
                prim->u0 = 0x80;
                prim->u3 = 0xFF;
                prim->u1 = 0xFF;
                prim->v1 = 0x80;
                prim->v0 = 0x80;
                prim->v3 = 0xBF;
                prim->v2 = 0xBF;
                prim->priority = 0x20;
                prim->blendMode = 2;
                prim = prim->next;
            } 
the if statement is not necessary, this is a common pattern for Entities like this, you will find it very often :)

Reply:
the if statement is not necessary, this is a common pattern for Entities like this, you will find it very often :)